ecr-login/config/registry/registry.go

```diff
@@ -0,0 +1,93 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+type RegistryConfig struct {
+    Profile string `yaml:"profile"`
+}
+
+type RegistryConfigs struct {
+    RegistryConfigs map[string]RegistryConfig `yaml:"registryConfigs"`
+}
+
+const ENV_AWS_ECR_REGISTRY_CONFIG_PATH = "AWS_ECR_REGISTRY_CONFIG_PATH"
+
+var (
+    RegistryConfigPath = getRegistryConfigPath()
+    RegistryConfigFilePath = filepath.Join(RegistryConfigPath, "registryConfig.yaml")
+    GetRegistryProfile = getRegistryProfile // Provide override for mocking
+)
+
+// Function to determine the RegistryConfigPath
+func getRegistryConfigPath() string {
+	// Get the path from the environment variable
+	path := os.Getenv(ENV_AWS_ECR_REGISTRY_CONFIG_PATH)
+	if path == "" {
+		expandedPath, err := os.UserHomeDir()
+		if err != nil {
+			expandedPath = "." // Fallback to the current directory if home directory cannot be resolved
```

Comment:
This could lead to some unexpected behavior. I would expect that the program proceed as if there is no config file in this case. Happy to be told that this is customary and expected though.